libs/go/aealite/protocols/dialogue.go

```diff
@@ -0,0 +1,142 @@
+package protocols
+
+import (
+	"fmt"
+	"crypto/rand"
+	"encoding/hex"
+)
+
+type Address string
+type Performative []string
+type SomeMessageType string
+
+const(
+	NONCE_BYTES_NB = 32
+)
+
+// type Dialogue struct {
+// 	_dialogues_storage
+// 	_dialogues_stats
+// 	_keep_terminal_state_dialogues
+// 	_message_class
+// 	_dialogue_class
+// 	_role_from_first_message
+// 	_self_address
+// 	_outgoing_messages
+// 	_incoming_messages
+// }
+
+// TODO
+// Define RoleType and figure out how to find role
+
+type Label struct {
```

Comment:
LEt's call it DialogueLabel for symmetry. I'd implement the method `get_incomplete_version` for it too. As well as the getters for starter and responder reference.